docs/01-introduction.Rmd

```diff
@@ -189,43 +162,39 @@ Try it again! Add another R code chunk like:
 ````
 ```{r echo=FALSE}`r ''`
 library(ggplot2)
-oplot <- ggplot(Orange, aes(x = age, 
+ggplot(Orange, aes(x = age, 
                    y = circumference, 
-                   colour = Tree)) +
+                   color = Tree)) +
   geom_point() +
   geom_line() +
-  guides(colour = FALSE) +
+  guides(color = FALSE) +
   theme_bw()
-oplot
 ```
 ````
 
-You should see a blog post that looks like this:
+You should see a blog post that looks like Figure \@ref(fig:edit-content).
 
 ```{r edit-content, fig.cap='Adding new R code to an .Rmd post.', echo=FALSE, fig.align='center', out.width='80%'}
 knitr::include_graphics('images/edit-content.png')
 ```
 
+### When in doubt, run `blogdown::check_site()`
 
-Any time you edit your website content, you may want to get in the habit of using one of the **blogdown** checking functions, which can help you identify problems with your content files:
+There are many possible reasons why a website does not work as expected. For example, you may have misconfigured certain options in the website configuration file (such as `config.yaml`), or marked a post as draft but forgetten to unmark it before publishing the site. It is easy to make these mistakes, and often hard to diagnose them.
 
-```{r eval=FALSE}
-blogdown::check_content()
-```
+Fortunately, the function `blogdown::check_site()` can run a series of checks in your website project to help you identify common problems. It also provides tips on how to fix these problems. It is meant to be used during your interactive work sessions. You will see what files were checked, which checks were successful, and **blogdown** will offer `[TODO]` items when a check needs your attention, like this:
 
-These functions are meant to be used during your interactive work sessions. You'll see what files were checked, which checks were successful, and **blogdown** will offer `[TODO]` items when a check needs your attention, like this:
-
-```
+```text
 --------------------------------------------------------------------
 ○ A successful check looks like this.
 ● [TODO] A check that needs your attention looks like this.
 | Let's check out your blogdown site!
 --------------------------------------------------------------------
 ```
 
-Checking the content of our example site should print output like this:
+For example, checking the content of our example site should print output like this:
 
-```
+```text
```

Comment:
Hi @yihui - the output here no longer really makes sense- I was trying to introduce the checking functions as they applied (so it wouldn't be overwhelming at first)- but this is just the output to `check_content()`- you may want to update with the whole output of `check_site()` I guess?

Reply:
Okay, I'll add more lines of output and omit some of them (by using `...`).

Here I avoided mentioning `check_content()` specifically because I think it's too much for readers to remember the five specific checking functions in Chapter 1 (especially in a section titled "A quick example"---we want to make it really quick). I hope that makes sense.

Reply:
That makes sense, we just disagree. I was trying to limit cognitive load b/c there is so much new vocab to learn, and I outlined the sections to map onto the functions to help build a mental model (I had hoped). 

But I'm ok with this change- all my later PRs followed the same model, so I'll let you make those adaptations as you go.

Reply:
I think we agree on the goal but disagree on the means. The goal of introducing `check_site()` instead of `check_content()` is meant exactly to limit cognitive load and new vocab. It's very unlikely that `check_content()` alone would help most users, and we will have to introduce `check_site()` at some point in the book, which means users will have to continue digging into the rest of the book, and remember at least two functions, `check_content()` and `check_site()`. To me, that requires more cognitive load and larger vocab.

To be honest, I was still not convinced that the sub functions of `check_site()` need to be exported and used individually. I can't even remember all of them easily myself, and wouldn't use them individually, because when my site is in trouble, it's difficult for me to predict which part went wrong (config? gitignore? content? netlify?), so it'd also be difficult for me to know which specific `check_*()` function to run. I think this applies to you, too (e.g. last Friday you didn't realize the problem was from `config.yaml`, so you wouldn't be able to know that you had to run `check_config()`; even after I told you it was because of `ignoreFiles`, you were trying to look at `.gitignore`---perhaps you misheard me). The downside of running `check_site()` is that it might output some noise (parts that are normal), but the noise can be relatively easily filtered out since the TODO items are prominently marked.

It makes perfect sense to explain what `check_site()` does in five parts, but it makes little sense to me to teach users to do these five steps separately. That said, I trust your expertise in teaching more than mine, so I exported these five functions anyway. In practice, I'd use the one and only function to check everything.